.github/workflows/build.yml

```diff
@@ -99,9 +103,17 @@ jobs:
 
   # Repackage separately for ESP82xx and ESP32
   repackage:
-    needs: build
+    needs: [documentation, generate-matrix]
```

Comment:
I think it needs build too?

Reply:
The intent is to initialize this job early, and not wait for `build` as it does currently, so it will get a runner assigned once that becomes available. But the culprit seems to be the matrix-build, causing the runner assignments to go haywire in some way.

Most likely this PR will eventually be abandoned, as all this doesn't seem to work as intended. I'll do a few more tests.

Reply:
NB: The first step in this job is to wait for the `build` job to complete, but it doesn't seem to find it.

Reply:
![image](https://user-images.githubusercontent.com/3751318/220269247-746c5672-e53f-4881-9b0d-2b898d071bad.png)
It isn't depending on it, so the build is probably not "in scope" of the repackage step.

Reply:
The action used here was designed to avoid the `needs` attribute for the dependency check and be an alternative for that, as by using `needs` will only create the job after `build` and request a runner for it. It will then be appended to the (already long) list of other jobs to be run, causing this job to execute very late, which is what I'm trying to solve here 🤔